src/playground/sample_data_ingestion/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+    """
+    Function that constructs the request body for BULK API call
+
+    Arguments:  
+        - index_name: The name of the index to ingest data
+        - file_provided: Boolean flag as to whether or not a file was provided
+        - chunk: How many documents can be ingested per BULK call
+        - timestamp: The field name that contains timestamps
+        - minutes: The time interval for each data point (e.g. if minutes = 2, this tool will generate entries with timestamps that are 2 minutes apart from one another)
+        - current_date: The date at which entries are generated
+        - dataset: The list containing the data to ingest
+        - max_bulk_size: The max amount in bytes of a bulk call
+
+    Returns:
+        - A tuple containing the request body and the current index to look at
+    """
+
+    one_week_ago = datetime.now() - timedelta(days = 7)
+    add_to_request = []
+    request_body = []
+    current_body_size = sys.getsizeof(request_body)
+    while current_index < min(chunk_index + chunk, len(dataset)):
+        # Adds the index name
+        index_name_body = {"index": {"_index": index_name}}
+
+        # If timestamps were a field type for user provided data, modify time
+        if file_provided and timestamp:
+            one_week_ago += timedelta(minutes = minutes)
+            offset = one_week_ago.strftime("%s")
+            dataset[current_index][timestamp] = int(offset)
+        
+        # Adds the action to take
+        index_action = dataset[current_index]
+
+        # Adds both index name body and index action body to the test request body
+        add_to_request.append(index_name_body)
+        add_to_request.append(index_action)
+        current_body_size = sys.getsizeof(add_to_request)
+
+        # Adds to request only if the body size is < the max bulk size (user defined)
+        if current_body_size < max_bulk_size:
+            request_body.append(index_name_body)
+            request_body.append(index_action)
+        else:
+            break
+        current_index+= 1
+    
+    return (request_body, current_index)
+
+
+def ingest(client, 
+    template,
+    index_name,
+    file_provided = False,  
+    mapping = True,  
+    number = 6, 
+    chunk = 5, 
+    timestamp = None,
+    feature = None,  
+    avg_min = 0, 
+    avg_max = 0, 
+    abs_min = 0, 
+    abs_max = 0, 
+    anomaly_percentage = 0, 
+    minutes = 2,
+    current_date = datetime.today(),
+    max_bulk_size = 100000
+):
+    """
+    Function that ingests user-provided data or generated data
+
+    Arguments:  
+        - client: an OpenSearch Python client object
+        - template: JSON file, CSV file, JSON mapping, or JSON short-hand template
+        - index_name: The name of the index to ingest data
+        - mapping: Whether or not the input is a mapping
+        - file_provided: Boolean flag as to whether or not a file was provided
+        - chunk: How many documents can be ingested per BULK call
```

Comment:
It's a good start! Non-blocking: Let's add some unit tests around the chunk and bulk size.

Reply:
Done, added a couple tests to the chunk and bulk size